src/streamsync/core.py

```diff
@@ -1022,9 +1038,13 @@ def _call_handler_callable(self, event_type, target_component, instance_path, pa
                 arg_values.append(session_info)
 
         result = None
-        with contextlib.redirect_stdout(io.StringIO()) as f:
-            result = callable_handler(*arg_values)
-        captured_stdout = f.getvalue()
+        if is_async_handler:
+            async_handler = self._async_handler_executor(callable_handler, arg_values)
+            thread_pool_future = thread_pool.submit(asyncio.run, async_handler)
+            result, captured_stdout = thread_pool_future.result()
```

Comment:
Following my proposal in EventHandler, here is the final code in ``_call_handler_callable``

```python
        if is_async_handler:
            result, captured_stdout = self._async_handler_executor(callable_handler, arg_values)
        else:
            result, captured_stdout = self._sync_handler_executor(callable_handler, arg_values)
```

Reply:
This does look cleaner indeed, thank you for your input. I'll implement it as per your suggestion.